hpx/runtime/applier/apply_helper.hpp

```diff
@@ -116,39 +115,78 @@ namespace hpx { namespace applier { namespace detail
         }
     };
 
+    ///////////////////////////////////////////////////////////////////////////
+    HPX_FORCEINLINE bool needs_to_execute_asynchronously()
```

Comment:
Not quite, I left those separate as I think we will end up with different criteria for deciding whether we have sufficient stack space.